### ISSUE TITLE
Database failure [TMS 222181] 

### ISSUE THREAD

**Issue by Pheotis** [stargate-rewritten/Stargate-Bukkit#193]
Saved as `vault_2` via core ftp, with database dump.

# On an imported knarvik instance running on a recent paper 1.18.2 build, on startup (first connection after fresh sql database connected):
## On startup
```java
[12:03:33 INFO]: [Stargate] Enabling Stargate v1.0.0.6-ALPHA
[12:03:33 WARN]: [Stargate] Invalid colors for sign text. Using default colors instead...
[12:03:33 INFO]: [Stargate] Saving language file from internal path lang\en-GB\en.txt
[12:03:33 INFO]: [Stargate] Saving language file from internal path lang\en\en-GB.txt
[12:03:33 INFO]: [Stargate] Saving language file from internal path lang\en\en.txt
[12:03:33 INFO]: [Stargate] Checking internal language file 'lang\en-GB\en-GB.txt'
[12:03:33 INFO]: [com.zaxxer.hikari.HikariDataSource] HikariPool-1 - Starting...
[12:03:36 INFO]: [com.zaxxer.hikari.HikariDataSource] HikariPool-1 - Start completed.
[12:03:39 WARN]: java.sql.SQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'IF NOT EXISTS SG_2537f21647234591a9ce9c711d7baf48PortalView AS SELECT SG_2537f21' at line 1
[12:03:39 WARN]:        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:120)
[12:03:39 WARN]:        at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
[12:03:39 WARN]:        at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:953)
[12:03:39 WARN]:        at com.mysql.cj.jdbc.ClientPreparedStatement.execute(ClientPreparedStatement.java:371)
[12:03:39 WARN]:        at Stargate-1.0.0.6-ALPHA.jar//com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
[12:03:39 WARN]:        at Stargate-1.0.0.6-ALPHA.jar//com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
[12:03:39 WARN]:        at Stargate-1.0.0.6-ALPHA.jar//net.TheDgtl.Stargate.database.PortalDatabaseAPI.runStatement(PortalDatabaseAPI.java:273)
[12:03:39 WARN]:        at Stargate-1.0.0.6-ALPHA.jar//net.TheDgtl.Stargate.database.PortalDatabaseAPI.createTables(PortalDatabaseAPI.java:310)
[12:03:39 WARN]:        at Stargate-1.0.0.6-ALPHA.jar//net.TheDgtl.Stargate.database.PortalDatabaseAPI.<init>(PortalDatabaseAPI.java:84)
[12:03:39 WARN]:        at Stargate-1.0.0.6-ALPHA.jar//net.TheDgtl.Stargate.database.PortalDatabaseAPI.<init>(PortalDatabaseAPI.java:62)
[12:03:39 WARN]:        at Stargate-1.0.0.6-ALPHA.jar//net.TheDgtl.Stargate.Stargate.load(Stargate.java:338)
[12:03:39 WARN]:        at Stargate-1.0.0.6-ALPHA.jar//net.TheDgtl.Stargate.Stargate.onEnable(Stargate.java:161)
[12:03:39 WARN]:        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264)
[12:03:39 WARN]:        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:370)
[12:03:39 WARN]:        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:541)
[12:03:39 WARN]:        at org.bukkit.craftbukkit.v1_18_R2.CraftServer.enablePlugin(CraftServer.java:560)
[12:03:39 WARN]:        at org.bukkit.craftbukkit.v1_18_R2.CraftServer.enablePlugins(CraftServer.java:474)
[12:03:39 WARN]:        at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:664)
[12:03:39 WARN]:        at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:431)
[12:03:39 WARN]:        at net.minecraft.server.dedicated.DedicatedServer.e(DedicatedServer.java:316)
[12:03:39 WARN]:        at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1163)
[12:03:39 WARN]:        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:315)
[12:03:39 WARN]:        at java.base/java.lang.Thread.run(Thread.java:833)
```
## On User Join
```java
[12:13:11 INFO]: UUID of player Central is 9e14ae1a-0823-35e1-9d4d-a3764171d58d
[12:13:12 INFO]: Central joined the game
[12:13:12 INFO]: Central[/127.0.0.1:49611] logged in with entity id 1 at ([world]38.097036930740096, -58.66773510344885, -32.34397679255828)
[12:13:12 WARN]: java.lang.NullPointerException: Cannot invoke "net.TheDgtl.Stargate.database.StorageAPI.startInterServerConnection()" because the return value of "net.TheDgtl.Stargate.Stargate.getStorageAPIStatic()" is null
[12:13:12 WARN]:        at Stargate-1.0.0.6-ALPHA.jar//net.TheDgtl.Stargate.listener.PlayerEventListener.lambda$2(PlayerEventListener.java:197)
[12:13:12 WARN]:        at Stargate-1.0.0.6-ALPHA.jar//net.TheDgtl.Stargate.action.ConditionalDelayedAction.run(ConditionalDelayedAction.java:32)
[12:13:12 WARN]:        at Stargate-1.0.0.6-ALPHA.jar//net.TheDgtl.Stargate.thread.SynchronousPopulator.cycleQueue(SynchronousPopulator.java:104)
[12:13:12 WARN]:        at Stargate-1.0.0.6-ALPHA.jar//net.TheDgtl.Stargate.thread.SynchronousPopulator.cycleQueues(SynchronousPopulator.java:83)
[12:13:12 WARN]:        at Stargate-1.0.0.6-ALPHA.jar//net.TheDgtl.Stargate.thread.SynchronousPopulator.run(SynchronousPopulator.java:38)
[12:13:12 WARN]:        at org.bukkit.craftbukkit.v1_18_R2.scheduler.CraftTask.run(CraftTask.java:101)
[12:13:12 WARN]:        at org.bukkit.craftbukkit.v1_18_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:483)
[12:13:12 WARN]:        at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1531)
[12:13:12 WARN]:        at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:483)
[12:13:12 WARN]:        at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1455)
[12:13:12 WARN]:        at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1225)
[12:13:12 WARN]:        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:315)
[12:13:12 WARN]:        at java.base/java.lang.Thread.run(Thread.java:833)
```
## On User Move
```java
[12:14:28 ERROR]: Could not pass event PlayerMoveEvent to Stargate v1.0.0.6-ALPHA
java.lang.NullPointerException: Cannot invoke "net.TheDgtl.Stargate.network.RegistryAPI.getPortal(org.bukkit.Location, net.TheDgtl.Stargate.gate.structure.GateStructureType)" because the return value of "net.TheDgtl.Stargate.Stargate.getRegistryStatic()" is null
        at net.TheDgtl.Stargate.listener.MoveEventListener.onAnyMove(MoveEventListener.java:99) ~[Stargate-1.0.0.6-ALPHA.jar:?]
        at net.TheDgtl.Stargate.listener.MoveEventListener.onPlayerMove(MoveEventListener.java:70) ~[Stargate-1.0.0.6-ALPHA.jar:?]
        at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor33.execute(Unknown Source) ~[?:?]
        at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:75) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:git-Paper-350]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:669) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at net.minecraft.server.network.ServerGamePacketListenerImpl.handleMovePlayer(ServerGamePacketListenerImpl.java:1504) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundMovePlayerPacket.handle(ServerboundMovePlayerPacket.java:29) ~[?:?]
        at net.minecraft.network.protocol.game.ServerboundMovePlayerPacket$Pos.handle(ServerboundMovePlayerPacket.java:66) ~[?:?]
        at net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$1(PacketUtils.java:51) ~[?:?]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.18.2.jar:git-Paper-350]
        at net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:153) ~[?:?]
        at net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:24) ~[?:?]
        at net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:1399) ~[paper-1.18.2.jar:git-Paper-350]
        at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:188) ~[paper-1.18.2.jar:git-Paper-350]
        at net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:126) ~[?:?]
        at net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:1376) ~[paper-1.18.2.jar:git-Paper-350]
        at net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:1369) ~[paper-1.18.2.jar:git-Paper-350]
        at net.minecraft.util.thread.BlockableEventLoop.managedBlock(BlockableEventLoop.java:136) ~[?:?]
        at net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:1347) ~[paper-1.18.2.jar:git-Paper-350]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1229) ~[paper-1.18.2.jar:git-Paper-350]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:315) ~[paper-1.18.2.jar:git-Paper-350]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
```
## On shutdown
```java
[12:15:54 INFO]: [Stargate] Disabling Stargate v1.0.0.6-ALPHA
[12:15:54 ERROR]: Error occurred while disabling Stargate v1.0.0.6-ALPHA (Is it up to date?)
java.lang.NullPointerException: Cannot invoke "net.TheDgtl.Stargate.network.StargateRegistry.getBungeeNetworkMap()" because "this.registry" is null
        at net.TheDgtl.Stargate.Stargate.onDisable(Stargate.java:349) ~[Stargate-1.0.0.6-ALPHA.jar:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:266) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.java.JavaPluginLoader.disablePlugin(JavaPluginLoader.java:399) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.disablePlugin(SimplePluginManager.java:578) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.disablePlugins(SimplePluginManager.java:555) ~[paper-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_18_R2.CraftServer.disablePlugins(CraftServer.java:494) ~[paper-1.18.2.jar:git-Paper-350]
        at net.minecraft.server.MinecraftServer.stopServer(MinecraftServer.java:971) ~[paper-1.18.2.jar:git-Paper-350]
        at net.minecraft.server.dedicated.DedicatedServer.stopServer(DedicatedServer.java:803) ~[paper-1.18.2.jar:git-Paper-350]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1270) ~[paper-1.18.2.jar:git-Paper-350]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:315) ~[paper-1.18.2.jar:git-Paper-350]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

# On an imported simulation of an old SGR config, running on a recent spigot:
## On Startup
```java
[12:04:10] [Server thread/INFO]: [Stargate] Enabling Stargate v1.0.0.6-ALPHA
[12:04:10] [Server thread/INFO]: [Stargate] Saving language file from internal path lang\en-CA\en.txt
[12:04:10] [Server thread/INFO]: [Stargate] Saving language file from internal path lang\en\en-CA.txt
[12:04:10] [Server thread/INFO]: [Stargate] Saving language file from internal path lang\en\en.txt
[12:04:10] [Server thread/INFO]: [Stargate] Checking internal language file 'lang\en-CA\en-CA.txt'
[12:04:10] [Server thread/INFO]: HikariPool-1 - Starting...
[12:04:11] [Server thread/INFO]: HikariPool-1 - Start completed.
[12:04:12] [Server thread/WARN]: java.sql.SQLSyntaxErrorException: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'IF NOT EXISTS SG_56a498063e1e454baa75e008d098d365PortalView AS SELECT SG_56a4980' at line 1
[12:04:12] [Server thread/WARN]:        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:120)
[12:04:12] [Server thread/WARN]:        at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
[12:04:12] [Server thread/WARN]:        at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:953)
[12:04:12] [Server thread/WARN]:        at com.mysql.cj.jdbc.ClientPreparedStatement.execute(ClientPreparedStatement.java:371)
[12:04:12] [Server thread/WARN]:        at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
[12:04:12] [Server thread/WARN]:        at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
[12:04:12] [Server thread/WARN]:        at net.TheDgtl.Stargate.database.PortalDatabaseAPI.runStatement(PortalDatabaseAPI.java:273)
[12:04:12] [Server thread/WARN]:        at net.TheDgtl.Stargate.database.PortalDatabaseAPI.createTables(PortalDatabaseAPI.java:310)
[12:04:12] [Server thread/WARN]:        at net.TheDgtl.Stargate.database.PortalDatabaseAPI.<init>(PortalDatabaseAPI.java:84)
[12:04:12] [Server thread/WARN]:        at net.TheDgtl.Stargate.database.PortalDatabaseAPI.<init>(PortalDatabaseAPI.java:62)
[12:04:12] [Server thread/WARN]:        at net.TheDgtl.Stargate.Stargate.load(Stargate.java:338)
[12:04:12] [Server thread/WARN]:        at net.TheDgtl.Stargate.Stargate.onEnable(Stargate.java:161)
[12:04:12] [Server thread/WARN]:        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:264)
[12:04:12] [Server thread/WARN]:        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:342)
[12:04:12] [Server thread/WARN]:        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:480)
[12:04:12] [Server thread/WARN]:        at org.bukkit.craftbukkit.v1_18_R2.CraftServer.enablePlugin(CraftServer.java:518)
[12:04:12] [Server thread/WARN]:        at org.bukkit.craftbukkit.v1_18_R2.CraftServer.enablePlugins(CraftServer.java:432)
[12:04:12] [Server thread/WARN]:        at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:612)
[12:04:12] [Server thread/WARN]:        at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:414)
[12:04:12] [Server thread/WARN]:        at net.minecraft.server.dedicated.DedicatedServer.e(DedicatedServer.java:263)
[12:04:12] [Server thread/WARN]:        at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1007)
[12:04:12] [Server thread/WARN]:        at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:304)
[12:04:12] [Server thread/WARN]:        at java.base/java.lang.Thread.run(Thread.java:833)
[12:04:12] [Server thread/INFO]: Server permissions file permissions.yml is empty, ignoring it
[12:04:12] [Server thread/INFO]: Done (11.374s)! For help, type "help"
```
## On user join
```java
[12:10:23] [User Authenticator #1/INFO]: UUID of player UserTest is 9b60a229-cefc-3097-8785-2af4931d7fcc
[12:10:24] [Server thread/INFO]: UserTest[/127.0.0.1:49577] logged in with entity id 1 at ([world]5.221439779654942, -59.76026256198606, -10.252175535683296)
[12:10:26] [Server thread/WARN]: java.lang.NullPointerException: Cannot invoke "net.TheDgtl.Stargate.database.StorageAPI.startInterServerConnection()" because the return value of "net.TheDgtl.Stargate.Stargate.getStorageAPIStatic()" is null
[12:10:26] [Server thread/WARN]:        at net.TheDgtl.Stargate.listener.PlayerEventListener.lambda$2(PlayerEventListener.java:197)
[12:10:26] [Server thread/WARN]:        at net.TheDgtl.Stargate.action.ConditionalDelayedAction.run(ConditionalDelayedAction.java:32)
[12:10:26] [Server thread/WARN]:        at net.TheDgtl.Stargate.thread.SynchronousPopulator.cycleQueue(SynchronousPopulator.java:104)
[12:10:26] [Server thread/WARN]:        at net.TheDgtl.Stargate.thread.SynchronousPopulator.cycleQueues(SynchronousPopulator.java:83)
[12:10:26] [Server thread/WARN]:        at net.TheDgtl.Stargate.thread.SynchronousPopulator.run(SynchronousPopulator.java:38)
[12:10:26] [Server thread/WARN]:        at org.bukkit.craftbukkit.v1_18_R2.scheduler.CraftTask.run(CraftTask.java:82)
[12:10:26] [Server thread/WARN]:        at org.bukkit.craftbukkit.v1_18_R2.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:415)
[12:10:26] [Server thread/WARN]:        at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1285)
[12:10:26] [Server thread/WARN]:        at net.minecraft.server.dedicated.DedicatedServer.b(DedicatedServer.java:429)
[12:10:26] [Server thread/WARN]:        at net.minecraft.server.MinecraftServer.a(MinecraftServer.java:1237)
[12:10:26] [Server thread/WARN]:        at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1047)
[12:10:26] [Server thread/WARN]:        at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:304)
[12:10:26] [Server thread/WARN]:        at java.base/java.lang.Thread.run(Thread.java:833)
```
## On shutdown
```java
[12:16:34] [Server thread/INFO]: [Stargate] Disabling Stargate v1.0.0.6-ALPHA
[12:16:34] [Server thread/ERROR]: Error occurred while disabling Stargate v1.0.0.6-ALPHA (Is it up to date?)
java.lang.NullPointerException: Cannot invoke "net.TheDgtl.Stargate.database.StorageAPI.endInterServerConnection()" because "this.storageAPI" is null
        at net.TheDgtl.Stargate.Stargate.onDisable(Stargate.java:367) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:266) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.java.JavaPluginLoader.disablePlugin(JavaPluginLoader.java:367) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.disablePlugin(SimplePluginManager.java:501) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.disablePlugins(SimplePluginManager.java:493) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_18_R2.CraftServer.disablePlugins(CraftServer.java:452) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3491-Spigot-b081915-95d233d]
        at net.minecraft.server.MinecraftServer.t(MinecraftServer.java:899) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3491-Spigot-b081915-95d233d]
        at net.minecraft.server.dedicated.DedicatedServer.t(DedicatedServer.java:717) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3491-Spigot-b081915-95d233d]
        at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1086) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3491-Spigot-b081915-95d233d]
        at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:304) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3491-Spigot-b081915-95d233d]
        at java.lang.Thread.run(Thread.java:833) [?:?]
```

# On an imported legacy version running on 1.15 craftbukkit:
## On startup
```java
[12:04:08] [Server thread/INFO]: [Stargate] Default stargate coloring is not supported on your current server implementation
[12:04:09] [Server thread/INFO]: [Stargate] Saving language file from internal path lang\en-GB\en.txt
[12:04:09] [Server thread/INFO]: [Stargate] Saving language file from internal path lang\en\en-GB.txt
[12:04:09] [Server thread/INFO]: [Stargate] Saving language file from internal path lang\en\en.txt
[12:04:09] [Server thread/INFO]: [Stargate] Checking internal language file 'lang\en-GB\en-GB.txt'
[12:04:09] [Server thread/ERROR]: Error occurred while enabling Stargate v1.0.0.6-ALPHA (Is it up to date?)
java.lang.NoClassDefFoundError: org/slf4j/LoggerFactory
        at com.zaxxer.hikari.HikariConfig.<clinit>(HikariConfig.java:51) ~[?:?]
        at net.TheDgtl.Stargate.database.MySqlDatabase.setupConfig(MySqlDatabase.java:95) ~[?:?]
        at net.TheDgtl.Stargate.database.MySqlDatabase.<init>(MySqlDatabase.java:46) ~[?:?]
        at net.TheDgtl.Stargate.database.PortalDatabaseAPI.loadDatabase(PortalDatabaseAPI.java:255) ~[?:?]
        at net.TheDgtl.Stargate.database.PortalDatabaseAPI.<init>(PortalDatabaseAPI.java:62) ~[?:?]
        at net.TheDgtl.Stargate.Stargate.load(Stargate.java:338) ~[?:?]
        at net.TheDgtl.Stargate.Stargate.onEnable(Stargate.java:161) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:352) [spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:417) [spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugin(CraftServer.java:462) [spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugins(CraftServer.java:376) [spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at net.minecraft.server.v1_15_R1.MinecraftServer.a(MinecraftServer.java:456) [spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at net.minecraft.server.v1_15_R1.DedicatedServer.init(DedicatedServer.java:266) [spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:783) [spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at java.lang.Thread.run(Thread.java:829) [?:?]
Caused by: java.lang.ClassNotFoundException: org.slf4j.LoggerFactory
        at java.net.URLClassLoader.findClass(URLClassLoader.java:476) ~[?:?]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:135) ~[spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:81) ~[spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:589) ~[?:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
        ... 16 more
```
## On shutdown
```java
[12:13:51] [Server thread/INFO]: [Stargate] Disabling Stargate v1.0.0.6-ALPHA
[12:13:51] [Server thread/ERROR]: Error occurred while disabling Stargate v1.0.0.6-ALPHA (Is it up to date?)
java.lang.NullPointerException: null
        at net.TheDgtl.Stargate.Stargate.onDisable(Stargate.java:349) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:265) ~[spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at org.bukkit.plugin.java.JavaPluginLoader.disablePlugin(JavaPluginLoader.java:377) [spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at org.bukkit.plugin.SimplePluginManager.disablePlugin(SimplePluginManager.java:438) [spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at org.bukkit.plugin.SimplePluginManager.disablePlugins(SimplePluginManager.java:430) [spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at org.bukkit.craftbukkit.v1_15_R1.CraftServer.disablePlugins(CraftServer.java:396) [spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at net.minecraft.server.v1_15_R1.MinecraftServer.stop(MinecraftServer.java:698) [spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at net.minecraft.server.v1_15_R1.DedicatedServer.stop(DedicatedServer.java:683) [spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:864) [spigot-1.15.jar:git-Spigot-3b314f5-e2790ae]
        at java.lang.Thread.run(Thread.java:829) [?:?]
```

**Comment by EpicKnarvik97** [comment]
It seems we use MariaDB syntax which isn't valid in MySQL. Only the startup logs are relevant. The others are just consequences of the StorageAPI crashing. This is basically the same as #167. I'm pretty sure the instance shouldn't matter as long as a MySQL database is used.

**Comment by EpicKnarvik97** [comment]
Once we've got the new database framework fully integrated, it should be a lot easier to just add queries that check for the existance of indexes and views before trying to create them, accounting for differences in syntax.

**Comment by EpicKnarvik97** [comment]
This might be relevant for the Craftbukkit problem: https://www.spigotmc.org/threads/hikaricp-no-class-found-loggerfactory.383574/
Or at least, it seems it's not properly shaded into the .jar, but why does this only happen on 1.15 CraftBukkit?

**Comment by Thorinwasher** [comment]
Why would that be the cause of a syntax error though?  I think we shade everything properly, as it's possible to see the actual code of hikaricp when looking into the .jar file

**Comment by EpicKnarvik97** [comment]
> Why would that be the cause of a syntax error though? I think we shade everything properly, as it's possible to see the actual code of hikaricp when looking into the .jar file

slf4j-api only being in scope test is bothering me a bit, as that basically means it won't be included in the compiled .jar. Not sure why this isn't normally a problem though.

**Comment by Thorinwasher** [comment]
I think it's just something hikari.cp uses for logging during tests

**Comment by EpicKnarvik97** [comment]
> I think it's just something hikari.cp uses for logging during tests

I'm not so sure: https://stackoverflow.com/questions/27175146/hikaricp-loggerfactory-class-definition-not-found

**Comment by Thorinwasher** [comment]
Okay, so it's not then. Let's change that in that case.

**Comment by EpicKnarvik97** [comment]
All three problems should be fixed with the merging of #199. This issue can be closed after verification.

**Comment by EpicKnarvik97** [comment]
The exact problems defined in this issue have been fixed.
